feat(java): Add Gradle sections describing dependencies metadata report (#5712)
getsentry/sentry-docs: src/platforms/java/common/gradle.mdx

---
title: Gradle
sidebar_order: 100
description: "Learn about using the Sentry Gradle Plugin."
---

The [Sentry Gradle Plugin](https://github.com/getsentry/sentry-android-gradle-plugin) is an addition to the main Java SDK and offers

> seamless integration with the Gradle build system. It supports the report of external dependencies of the application as part of events.

Author: Can we rephrase it somehow so it's open for adding new features to it? Dependency report generation is just of the features, so I'd like to phrase it somehow to leave room for future features, because the plugin is capable of more things.

> 
> The [Sentry Gradle Plugin](https://github.com/getsentry/sentry-android-gradle-plugin) is an addition to the main Java SDK and offers
> seamless integration with the Gradle build system. It supports the report of external dependencies of the application as part of events.

Reviewer: What about?
```suggestion

The [Sentry Gradle Plugin](https://github.com/getsentry/sentry-android-gradle-plugin) is an addition to the main Java SDK and offers
seamless integration with the Gradle build system. It supports the reporting of your application's external dependencies as part of events.
```

> seamless integration with the Gradle build system. It supports the report of external dependencies of the application as part of events.

Reviewer: Just reminder that whatever phrasing we land on here needs to also be applied [here](https://github.com/getsentry/sentry-docs/pull/5712/files#diff-f05ad6205f886c7271d791282e44b2b79dc36f05e08c1784e9a31c0025b3ba7cR76).

## Setup

### Install

Using Gradle in your application module's `build.gradle` add:

```groovy
buildscript {
    repositories {
        mavenCentral()
    }
}

plugins {
    id "io.sentry.android.gradle" version "{{ packages.version('sentry.java.android.gradle-plugin', '3.0.0') }}"
}
```

```kotlin
buildscript {
    repositories {
        mavenCentral()
    }
}

plugins {
    id("io.sentry.android.gradle") version "{{ packages.version('sentry.java.android.gradle-plugin', '3.0.0') }}"
}
```

<Note>

> Although the plugin id contains `android`, it will not do anything for a non-android project, besides collecting external dependencies.

Reviewer 2: ```suggestion
Although the plugin id contains `android`, it works for non-android projects as well, but with limited feature support (Right now: collecting external dependencies).
```

Reviewer: @romtsn & @markushi - you're emphasizing two different things with your versions. What's important here? That it can work for non-android projects or that it barely works for non-android projects? @lizokm and I can make suggestions once we know what's most important to communicate.

Author: Important is that it *can* work for non-android projects, but what's even more important, that it doesn't do any unnecessary work for non-android projects

Reviewer: Ah gotcha. Okay what about:
```suggestion
Although the plugin id contains `android`, it can be used for non-android projects. However, for non-android projects, the plugin only collects external dependencies.
```

Author: Yeah, that looks betterm thank you 🙏

</Note>

### Configure

We expose the following configuration values directly in `build.gradle`:

```groovy
sentry {
    // Disables or enables dependencies metadata reporting for Sentry.
    // If enabled, the plugin will collect external dependencies and
    // upload them to Sentry as part of events. If disabled, all the logic
    // related to the dependencies metadata report will be excluded.
    //
    // Default is enabled.
    includeDependenciesReport = true
}
```

```kotlin
sentry {
    // Disables or enables dependencies metadata reporting for Sentry.
    // If enabled, the plugin will collect external dependencies and 
    // upload them to Sentry as part of events. If disabled, all the logic
    // related to the dependencies metadata report will be excluded.
    //
    // Default is enabled.
    includeDependenciesReport.set(true)
}
```

> ## Dependencies report

Reviewer: Almost forgot, we use AP style title case

```suggestion
## Dependencies Report
```

The plugin automatically collects external dependencies (including transitive ones) of the application and makes a report available to the runtime; the Sentry Java SDK picks the report up at runtime and sends it along with every Sentry event. The event dependencies metadata will look similar to this:

> 

Author: It's actually not a runtime-, but a compile-time report, and then it gets ingested into runtime, so that the SDK can pick it up and send it with the event, so I'm not sure if this is correct

> The plugin automatically collects external dependencies (including transitive ones) of the application and makes a report available to the runtime; the Sentry Java SDK picks the report up at runtime and sends it along with every Sentry event. The event dependencies metadata will look similar to this:
> 

Reviewer 3: @romtsn Ah! Got it. How about: 
```suggestion
The plugin automatically collects an application's external dependencies (including transitive ones), and generates a compile-time report. The Sentry Java SDK picks up the report at runtime and sends it along with every Sentry event. The event dependencies metadata will look something like this:

```

> 

Author: Yeah, that looks good, thank you!
